Improve course breadcrumbs

andy-armstrong opened the pull request:
## [LEARNER-877](https://openedx.atlassian.net/browse/LEARNER-877)

### Description

https://openedx.atlassian.net/browse/LEARNER-877

This change improves the breadcrumbs shown at the top of a course. Now each individual breadcrumb item is a link (except for the last one) as shown in this screenshot:

![image](https://cloud.githubusercontent.com/assets/5985072/26606517/3e481800-4560-11e7-9bd9-0818d1e8b32c.png)

### Sandbox
- [x] https://andy-armstrong.sandbox.edx.org/courses/course-v1:edX+DemoX+Demo_Course/course/

Note: staff has the unified course/home experience, non-staff do not.

### Testing
- [ ] i18n
- [ ] RTL
- [ ] safecommit violation code review process
- [ ] Unit, integration, acceptance tests as appropriate

FYI: @lizcohen @marcotuts @catong 

### Post-review
- [ ] Rebase and squash commits

andy-armstrong commented:
@lizcohen @marcotuts @catong I was thinking more about the breadcrumbs so wanted to run some thoughts by you:

1. It seems redundant to have the unit name shown in the breadcrumb and as a heading. Given this, I'm thinking of dropping it.

2. If we do this, then the next question is whether the subsection should be linked. In some ways you are already on the subsection page, so I'm leaning towards not linking it too.

3. This only leave the course and the section as links. The first will go to the top of the course home page, and the second will link to an anchor for the section.

4. I was thinking about alternatives for "Course" as the label for the main tab. It mirrors the tab name which seems consistent, but I'm not sure where the user would expect a "Course" link to go to. I think of it as the course home page, but I'm not sure if that's the terminology we will be using in the documentation. Would it be clearer to have the label be "Course Home"? Another option would be to show the course's title, e.g. "edX Demonstration Course" or "edX Demonstration Course Home". I'll leave it as "Course" for now, but let me know what you think. Note that if we change this then we will change the breadcrumbs in all of the course-related pages (currently breadcrumbs and updates, and then soon search results).

catong commented:
@andy-armstrong These are nice improvements!

1) I think removing the unit name from the breadcrumb is a good idea. Could be just me, but I've found it disorienting to click a subsection in the course outline, and then not see that subsection name at the place I arrive at. So, now, with the subsection name being at the end of the breadcrumb trail, it's a good confirmation that yes, I'm at the subsection that I clicked on.

2) IMO it would be useful to link the subsection too, and have that link take me back to the outline but with focus at the subsection I just left. Andy, I think you showed me that? Or maybe it was the section.

4) I like the idea of having the course title as the base of the breadcrumb. If not, then perhaps "Course Outline"?

andy-armstrong commented:
Thanks, @catong. I've implemented the breadcrumbs as we discussed and I've pushed it up to my sandbox.

dianakhuang commented on lms/templates/courseware/courseware.html:
```diff
@@ -155,11 +155,31 @@
         <main id="main" tabindex="-1" aria-label="Content">
-            <div
-              class="path"
-              data-unified-course-view="${'true' if waffle.flag_is_active(request, UNIFIED_COURSE_VIEW_FLAG) else 'false'}"
```
Did we want to remove this waffle check? I don't see this logic elsewhere.

andy-armstrong replied:
Nice catch. I've added this back.

dianakhuang commented on lms/templates/courseware/courseware.html:
```diff
@@ -155,11 +155,31 @@
+                            % if chapter:
+                                <span class="icon fa fa-angle-right" aria-hidden="true"></span>
+                                <span class="nav-item">
+                                    <a href="/courses/course-v1:edX+DemoX+Demo_Course/course/#${unicode(chapter.location)}">${chapter.display_name}</a>
```
I imagine we don't want to hardcode the Demo Course URL here. 😀

andy-armstrong replied:
Whoops! Fixed.

andy-armstrong commented:
@dianakhuang @marcotuts @lizcohen I've addressed the review feedback so far, and in particular I've changed the first link in the breadcrumb to be the course title as recommended by Carol. 

![image](https://cloud.githubusercontent.com/assets/5985072/26326802/d086d4ce-3f0a-11e7-99bc-c483d3cb311c.png)

This is ready for re-review now, thanks!

andy-armstrong commented:
FYI @HarryRein

andy-armstrong commented:
@marcotuts @lizcohen As requested, I've reverted the UX back to the original design:
 - the breadcrumb has a link to the subsection, and renders the unit title
 - the unit title updates as the user navigates with a subsection
 - the start of the breadcrumb now says "Course" again

I've updated the sandbox with these changes. Please do the final review so that this can merge.

andy-armstrong commented:
FYI, here's the request from @marcotuts to revert those changes, as context for others on this thread:

> [10:12 AM] Marco Morales: we actually specifically had tests with and without the unit page.
> [10:12 AM] Marco Morales: and the ones with the unit page were a lot clearer for learners
> [10:12 AM] Marco Morales: as it helped them link their current page location to the page title.
> [10:13 AM] Andy Armstrong: I'll put these on the PR for context for anyone reading it
...
> [10:14 AM] Marco Morales: Unfortunately here I'd like to ask us to revert until we see evidence this change is helpful, since we know the tests actually showed a demonstrable improvement when we added that unit label.
> [10:15 AM] Marco Morales: I can justify / sell to partners that 3 weeks after our initial launch we killed the outline label / button / action
> [10:15 AM] Marco Morales: but I have a harder time justifying the other changes proposed in the PR given our earlier testing.

lizcohen commented:
This looks good except for one small issue. In sections beyond example week 1, the breadcrumb links are the wrong color, they are darker for some reason. 
<img width="762" alt="screen shot 2017-05-31 at 12 56 35 pm" src="https://cloud.githubusercontent.com/assets/19694600/26643863/c13622ca-4600-11e7-8aed-bfce3fd52445.png">
<img width="829" alt="screen shot 2017-05-31 at 12 56 48 pm" src="https://cloud.githubusercontent.com/assets/19694600/26643864/c140f204-4600-11e7-93f7-c49019577650.png">

andy-armstrong commented:
Great catch, @lizcohen. I've fixed the sass to use the same style for visited and unvisited links. I've also introduced the standard darker hover color so it is clear that the link is active.

lizcohen commented:
thanks, @andy-armstrong! 👍

catong commented:
👍
